src/sema/types.rs

```diff
@@ -1089,6 +1089,7 @@ impl Type {
                 Box::new(Type::Array(ty.clone(), dim[..dim.len() - 1].to_vec())),
             ),
             Type::Array(ty, dim) if dim.len() == 1 => Type::StorageRef(false, ty.clone()),
+            Type::Bytes(n) => Type::Bytes(*n),
```

Comment:
Bytes is not an array type.

Reply:
The code path leading to the crash is due to an array subscript, and this is valid in `solc` as well, so why is this not an array?

Reply:
In emit, `Type::Bytes` is an integer. The function you modified is also used in `emit` at the Solana target. If this is the solution for the bug, the emit cannot call this function.

Reply:
> The function you modified is also used in `emit` at the Solana target. If this is the solution for the bug, the emit cannot call this function.

I just noticed this as well, gotta find another approach